libraries/chain/include/eosio/chain/webassembly/interface.hpp

```diff
@@ -0,0 +1,581 @@
+         void __fixunstfti(uint128_t&, uint64_t, uint64_t) const;
+         void __fixsfti(int128_t&, float) const;
+         void __fixdfti(int128_t&, double) const;
+         void __fixunssfti(uint128_t&, float) const;
+         void __fixunsdfti(uint128_t&, double) const;
+         double __floatsidf(int32_t) const;
+         void __floatsitf(float128_t&, int32_t) const;
+         void __floatditf(float128_t&, uint64_t) const;
+         void __floatunsitf(float128_t&, uint32_t) const;
+         void __floatunditf(float128_t&, uint64_t) const;
+         double __floattidf(uint64_t, uint64_t) const;
+         double __floatuntidf(uint64_t, uint64_t) const;
+         int32_t __cmptf2(uint64_t, uint64_t, uint64_t, uint64_t) const;
+         int32_t __eqtf2(uint64_t, uint64_t, uint64_t, uint64_t) const;
+         int32_t __netf2(uint64_t, uint64_t, uint64_t, uint64_t) const;
+         int32_t __getf2(uint64_t, uint64_t, uint64_t, uint64_t) const;
+         int32_t __gttf2(uint64_t, uint64_t, uint64_t, uint64_t) const;
+         int32_t __letf2(uint64_t, uint64_t, uint64_t, uint64_t) const;
+         int32_t __lttf2(uint64_t, uint64_t, uint64_t, uint64_t) const;
+         int32_t __unordtf2(uint64_t, uint64_t, uint64_t, uint64_t) const;
+
+         REGISTER_CF_HOST_FUNCTION(__ashlti3);
+         REGISTER_CF_HOST_FUNCTION(__ashrti3);
+         REGISTER_CF_HOST_FUNCTION(__lshlti3);
+         REGISTER_CF_HOST_FUNCTION(__lshrti3);
+         REGISTER_CF_HOST_FUNCTION(__divti3);
+         REGISTER_CF_HOST_FUNCTION(__udivti3);
+         REGISTER_CF_HOST_FUNCTION(__multi3);
+         REGISTER_CF_HOST_FUNCTION(__modti3);
+         REGISTER_CF_HOST_FUNCTION(__umodti3);
+         REGISTER_CF_HOST_FUNCTION(__addtf3);
+         REGISTER_CF_HOST_FUNCTION(__subtf3);
+         REGISTER_CF_HOST_FUNCTION(__multf3);
+         REGISTER_CF_HOST_FUNCTION(__divtf3);
+         REGISTER_CF_HOST_FUNCTION(__negtf2);
+         REGISTER_CF_HOST_FUNCTION(__extendsftf2);
+         REGISTER_CF_HOST_FUNCTION(__extenddftf2);
+         REGISTER_CF_HOST_FUNCTION(__trunctfdf2);
+         REGISTER_CF_HOST_FUNCTION(__trunctfsf2);
+         REGISTER_CF_HOST_FUNCTION(__fixtfsi);
+         REGISTER_CF_HOST_FUNCTION(__fixtfdi);
+         REGISTER_CF_HOST_FUNCTION(__fixtfti);
+         REGISTER_CF_HOST_FUNCTION(__fixunstfsi);
+         REGISTER_CF_HOST_FUNCTION(__fixunstfdi);
+         REGISTER_CF_HOST_FUNCTION(__fixunstfti);
+         REGISTER_CF_HOST_FUNCTION(__fixsfti);
+         REGISTER_CF_HOST_FUNCTION(__fixdfti);
+         REGISTER_CF_HOST_FUNCTION(__fixunssfti);
+         REGISTER_CF_HOST_FUNCTION(__fixunsdfti);
+         REGISTER_CF_HOST_FUNCTION(__floatsidf);
+         REGISTER_CF_HOST_FUNCTION(__floatsitf);
+         REGISTER_CF_HOST_FUNCTION(__floatditf);
+         REGISTER_CF_HOST_FUNCTION(__floatunsitf);
+         REGISTER_CF_HOST_FUNCTION(__floatunditf);
+         REGISTER_CF_HOST_FUNCTION(__floattidf);
+         REGISTER_CF_HOST_FUNCTION(__floatuntidf);
```

Comment:
Again, concerns about reference types apply to most of the compiler builtin host functions above.